examples/src/main/scala/io/laserdisc/kafka/SafeGDPR1.scala

```diff
@@ -0,0 +1,58 @@
+package io.laserdisc.kafka
+
+import org.apache.kafka.streams.scala.{ Serdes, StreamsBuilder }
+import org.apache.kafka.streams.{ StreamsConfig, Topology }
+import zio.UIO
+import zio.kafka.streams.testkit.ZTestTopology
+import zio.kafka.streams.{ KafkaStreamsTopology, SafeTopic, TopologySdk }
+import zio.prelude.State
+import zio.test.Assertion.equalTo
+import zio.test.environment.TestEnvironment
+import zio.test.{ DefaultRunnableSpec, TestFailure, ZSpec, suite, testM, _ }
+
+import java.util.Properties
+
+object SafeGDPR1 extends TopologySdk {
+  val config: Properties = {
+    val jProperties = new Properties()
+    jProperties.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092")
+    jProperties
+  }
+
+  val passwordFilter: (String, String) => Boolean = (k, _) => k.startsWith("p")
+  val usernameFilter: (String, String) => Boolean = (k, _) => k.startsWith("u")
+
+  val newUsersIn: SafeTopic[String, String]  = SafeTopic("new-users-input", Serdes.String, Serdes.String)
+  val newUsersOut: SafeTopic[String, String] = SafeTopic("new-users-output", Serdes.String, Serdes.String)
+
+  val topologyReader: State[StreamsBuilder, Topology] = for {
+    input    <- streamFromSource(newUsersIn)
+    branches <- branch(input)(usernameFilter, passwordFilter)
+    _        <- streamSinkTo(branches(passwordFilter), newUsersOut)
```

Comment:
This is a bit better because I can clearly see the name `passwordFilter` for my branch. It's quite hard to make a mistake here, but if we do the compile will be happy and we still write password in clear text.